### PR TITLE
chore: adopt unary interceptors for stream calls

### PR DESCRIPTION
**Pull request by MaxSchaefer**
I've abstracted all the interceptors logic away from just being used for `Unary`. In preparation for https://github.com/helpwave/services/pull/859.

### CODE REVIEW

File: libs/common/hwgrpc/wrapper_stream.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package hwgrpc
+
+import (
+	"context"
+	"google.golang.org/grpc"
+)
+
+type WrapperStream struct {
```
[the remark this answers is not in the document]
Comment (FoseFx) [reply libs/common/hwgrpc/wrapper_stream.go:8]: Or is it a ContextualizedStream?

File: libs/common/auth/auth.go [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+
+func GetOAuthClientId() string {
+	clientId := hwutil.GetEnvOr("OAUTH_CLIENT_ID", DEFAULT_OAUTH_CLIENT_ID)
+	if clientId != DEFAULT_OAUTH_CLIENT_ID {
+		zlog.Warn().Str("OAUTH_CLIENT_ID", clientId).Msg("using custom OAuth client id")
+	}
+	return clientId
+}
+
+type OrganizationTokenClaim struct {
```
[the remark this answers is not in the document]
Comment (MaxSchaefer) [reply libs/common/auth/auth.go:97]: Jup. The file was shaken a little during refactoring. I have sorted it.

File: libs/common/auth/auth.go [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+	}
+	return Verifier
+}
+
+// IsAuthSetUp is true, GetOAuthConfig and GetIDTokenVerifier can be accessed safely
+func IsAuthSetUp() bool {
+	return OauthConfig != nil && Verifier != nil
+}
+
+func GetAuthCodeURL(ctx context.Context) string {
```
[the remark this answers is not in the document]
Comment (MaxSchaefer) [reply libs/common/auth/auth.go:155]: _Removed_ instead. Not in use.

File: libs/common/auth/auth.go [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+	if !ok {
+		return uuid.UUID{}, status.Error(codes.Internal, "organizationID not in context, set up auth")
+	} else {
+		return res, nil
+	}
+}
+
+// OrganizationIDFromMD retrieves the user defined organizationID
+// from the metadata of the request
+func OrganizationIDFromMD(ctx context.Context) (string, error) {
```
[the remark this answers is not in the document]
Comment (MaxSchaefer) [reply libs/common/auth/auth.go:223]: _Removed_

File: libs/common/hwgrpc/span_interceptor.go [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+	ctx := spanInterceptorPreNext(stream.Context())
+	stream = NewWrapperStream(stream, WithContext(ctx))
+	err := next(req, stream)
+	spanInterceptorPostNext(ctx, err)
+	return err
+}
+
+// handlerSpanInterceptor opens and closes a span around the next in the chain
+// it should only be used as the last element in the interceptor chain before the handler
+func spanInterceptorPreNext(ctx context.Context) context.Context {
```
[the remark this answers is not in the document]
Comment (MaxSchaefer) [reply libs/common/hwgrpc/span_interceptor.go:28]: Yes. Not pretty but okay. PreNext and PostNext are required due to our calls to next() that depends on Unary or Stream.

File: libs/common/hwgrpc/wrapper_stream.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (MaxSchaefer) [reply libs/common/hwgrpc/wrapper_stream.go:8]: ... dont let us dig too deep into english grammmer ... :D adjective-noun-compound? Idk ...
I've switched to https://github.com/grpc-ecosystem/go-grpc-middleware/blob/main/wrappers.go

File: libs/common/auth/auth.go [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+	Name string `json:"name"`
+
+	// Subject: preferred_username
+	PreferredUsername string `json:"preferred_username"`
+
+	// Subject: organization
+	Organization *OrganizationTokenClaim `json:"organization"`
+}
+
+func (t IDTokenClaims) AsExpected() error {
```
[the remark this answers is not in the document]
Comment (MaxSchaefer) [reply libs/common/auth/auth.go:94]: Yes https://github.com/helpwave/services/issues/866

**Review by FoseFx — APPROVED** [review]

**Pull request merged**